docs/ShapeInference.md

```diff
@@ -1,5 +1,20 @@
 <!--- SPDX-License-Identifier: Apache-2.0 -->
 
+# Background
```

Comment:
Only one main title or several?

Reply:
Made into a sub-section and moved it, thanks!